1. Basics/task_4/hw4.php

```diff
@@ -0,0 +1,362 @@
+ * 
+ * @param string    $requestMethod
+ * @param string    $requestUri
+ * @param array     $requestHeaders
+ * @param array     $requestBody
+ * @return array
+ */
+function getResponse($requestMethod, $requestUri, $requestHeaders, $requestBody) : array {
+    
+    try{
+        checkRequestMethod($requestMethod);
+        checkRequestUri($requestUri);
+        checkRequestContentTypeValue($requestHeaders["Content-Type"]);
+        checkRequestBody($requestBody);
+        checkFileExistance(FILE_NAME);
+
+        $codeAndMessage = getAccess($requestBody);
+        $statusCode = $codeAndMessage['statusCode'];
+        $bodyMessage = $codeAndMessage['bodyMessage'];
+    }
+    catch(Exception $ex) {
+        $statusCode = $ex->getCode();
+        $bodyMessage = $ex->getMessage();
+    };
+
+    $statusMessage = getResponseStatusMessage($statusCode);
+    $body = getResponseBody($statusCode, $bodyMessage);
+    $headers = [
+        "Date" => date(DATE_RFC1123),                           
+        "Server" => "Apache/2.2.14 (Win32)",                
+        "Content-Length" => strlen($body), 
+        "Connection" => "Closed",          
+        "Content-Type" => "text/html; charset=utf-8",       
+    ];
+        
+    return [
+        'statusCode' => $statusCode,
+        'statusMessage' => $statusMessage,
+        'headers' => $headers,
+        'body' => $body,
+    ];
+}
+
+/**
+ * @param string $method
+ * @throws Exception
+ * @return void
+ */
+function checkRequestMethod(string $method) : void {
+    if( $method === "POST") return;
+
+    throw new Exception("Method is invalid. The method POST is needed.", 400);
+}
+
+/**
+ * @param string $uri
+ * @throws Exception
+ * @return void
+ */
+function checkRequestUri(string $uri) : void {
+    if($uri === "/api/checkLoginAndPassword") return;
+
+    throw new Exception("URI is invalid.", 400);
+}
+
+/**
+ * @param string $contentType
+ * @throws Exception
+ * @return void
+ */
+function checkRequestContentTypeValue(string $contentType) : void {
+    if($contentType === "application/x-www-form-urlencoded") return;
+    
+    throw new Exception("Header's 'Content-Type' value is invalid.", 400);
+}
+
+/**
+ * @param array $body
+ * @throws Exception
+ * @return void
+ */
+function checkRequestBody(array $body) : void {
+    if(array_key_exists("login", $body) && array_key_exists("password", $body)) return;
+    
+    throw new Exception("Request body doesn't content 'login' or/and 'password' variable(s)", 400);
+}
+
+/**
+ * @param string $file
+ * @throws Exception
+ * @return void
+ */
+function checkFileExistance(string $file) : void {
+    if(file_exists($file)) return;
+    
+    throw new Exception('File ' . $file . ' doesn`t exist', 500);
+}
+
+/**
+ * Grants or denies access. Returns the result code and message.
+ * @param array $body
+ * @throws Exception
+ * @return array
+ */
+function getAccess($body) : array {
```

Comment:
not obvious what this function is doing, should be renamed

Reply:
Done. Now its name is ```checkLoginAndPassword```